website/src/components/Visualisations/IIGOPayments/Util/ProcessedPaymentsData.ts

```diff
@@ -0,0 +1,116 @@
+import { OutputJSONType, TeamName } from '../../../../consts/types'
+import { ProcessedTaxData, TaxData } from './IIGOPaymentsTypes'
+
+export const processPaymentsData = (data: OutputJSONType): ProcessedTaxData => {
+    if (data.GameStates.length === 0) return []
+
+    const retData: ProcessedTaxData = []
+    const IIGOHistory = Object.entries(
+        data.GameStates[data.GameStates.length - 1].IIGOHistory
+    )
+
+    const empty = (): TaxData => {
+        return {
+            expectedAlloc: 0,
+            actualAlloc: 0,
+            expectedTax: 0,
+            actualTax: 0,
+            expectedSanction: 0,
+            actualSanction: 0,
+        }
+    }
+
+    const TaxInfo: Map<TeamName, TaxData> = new Map<TeamName, TaxData>([
+        [1, empty()],
+        [2, empty()],
+        [3, empty()],
+        [4, empty()],
+        [5, empty()],
+        [6, empty()],
+    ])
+
+    IIGOHistory.forEach(([turnNumber, exchanges]) => {
+        if (exchanges) {
```

Comment:
there would be an undefined exchange? :O

Reply:
to be honest, I am not sure either but it is done this way [here](https://github.com/SOMAS2020/SOMAS2020/blob/main/website/src/components/Visualisations/Transactions/Util/ProcessTransactionData.ts#L15)  so I just follow

Reply:
ah ok :O